internal/dnfjson/cache.go

```diff
@@ -6,11 +6,15 @@ import (
 	"os"
 	"path/filepath"
 	"sort"
+	"sync"
 	"time"
 
 	"github.com/gobwas/glob"
 )
 
+// global cache locker
+var cacheLocks sync.Map
```

Comment:
This sounds like it would make things overly complicated (and indeed would make my suggestions about the update logic not work). Why don't we just require there only to be one cache for a given root at any given time?

Reply:
(considering this would anyway only work if there were several in the same process, in that case we could just share the object, no?)

Reply:
Sharing the entire object instead of just the lock does sound like a better idea.  We'd still need to maintain the same kind of map, only it would give you the whole `rpmCache` instead of just the lock.